mysql/check.py

```diff
@@ -847,11 +858,17 @@ def _is_innodb_engine_enabled(self, db):
             self.warning("Possibly innodb stats unavailable - error querying engines table: %s" % str(e))
             return False
 
-    def _get_replica_stats(self, db):
+    def _get_replica_stats(self, db, is_mariadb, replication_channel):
         try:
             with closing(db.cursor(pymysql.cursors.DictCursor)) as cursor:
                 replica_results = {}
-                cursor.execute("SHOW SLAVE STATUS;")
+                if is_mariadb and replication_channel:
+                    cursor.execute("SET @@default_master_connection = '{0}';".format(replication_channel))
```

Comment:
I think this is fine, but typically cursors shouldn't be recycled like this. I'm wondering if we shouldn't maybe do it like this just for best-practices:
```
with closing(db.cursor(pymysql.cursors.DictCursor)) as cursor:
    if is_mariadb and replication_channel:
        cursor.execute("SET @@default_master_connection = '{0}';".format(replication_channel))

with closing(db.cursor(pymysql.cursors.DictCursor)) as cursor:
    if not is_mariadb and replication_channel:
        cursor.execute("SHOW SLAVE STATUS FOR CHANNEL '{0}';".format(replication_channel))
    else:
        cursor.execute("SHOW SLAVE STATUS;")
``` 
(or similar)

what do you think?

Reply:
I'm not sure, is closing the cursor the same as closing the current session? 

`SET @@default_master_connection` only affects current session and in mariadb it has to be executed in the same session in which `SHOW SLAVE STATUS` is executed.